StakingERC721/contracts/StakingERC721.sol

```diff
@@ -0,0 +1,371 @@
+  /**
+   * Emited when user unstakes his/her staked token
+   */
+  event Unstaked(address indexed account, uint256 amount, uint256[] tokenIds);
+  /**
+   * Emitted when user claim his/her rewards token
+   */
+  event Claimed(address indexed account, uint256 amount);
+  /**
+   * Emitted when owner funds rewards token and restart rewarding session
+   */
+  event Funded(uint256 amount);
+  /**
+   * Emitted when rewards duration has been updated
+   */
+  event RewardsDurationUpdated(uint256 duration);
+  /**
+   * Emitted when owner recovered tokens from this contract
+   */
+  event Recovered(address token, uint256 amount);
+
+  /* -------------------------------------------------------------------------- */
+  /*                                  Modifier                                  */
+  /* -------------------------------------------------------------------------- */
+
+  modifier updateReward(address account) {
+    rewardPerTokenStored = _rewardPerToken();
+    lastUpdateTime = _lastTimeRewardApplicable();
+    if (account != address(0)) {
+      rewards[account] = _earned(account);
+      userRewardPerTokenPaid[account] = rewardPerTokenStored;
+    }
+    _;
+  }
+
+  /* -------------------------------------------------------------------------- */
+  /*                             External Functions                             */
+  /* -------------------------------------------------------------------------- */
+
+  /**
+   * @dev Constructor
+   * @param rewardsToken_ Address to rewards ERC20 token
+   * @param stakingToken_ Address to staking ERC20 token
+   * @param duration Reward duration in seconds
+   */
+  constructor(address rewardsToken_, address stakingToken_, uint256 duration) {
+    rewardsToken = IERC20(rewardsToken_);
+    stakingToken = IERC721(stakingToken_);
+    rewardsDuration = duration;
+  }
+
+  function connectToOtherContracts(
+    address[] memory otherContracts
+  ) external override onlyOwner {}
+
+  /**
+   * @notice Pause or unpause contract
+   * @dev Callable by owner
+   * @param newPaused Flag to new paused state
+   */
+  function setPaused(bool newPaused) external onlyOwner {
+    if (newPaused) {
+      _pause();
+    } else {
+      _unpause();
+    }
+  }
+
+  /**
+   * @notice Set rewards duration, only available to set after finish
+   * of previous rewards period.
+   * @dev Callable by owner
+   * @param duration New rewards duration
+   */
+  function setRewardsDuration(uint256 duration) external onlyOwner {
+    if (block.timestamp < periodFinish) {
+      revert WaitToFinish();
+    }
+    rewardsDuration = duration;
+    emit RewardsDurationUpdated(duration);
+  }
+
+  /**
+   * @notice Stakes user's NFTs
+   * @dev Callable only when unpaused
+   * @param tokenIds The tokenIds of the NFTs which will be staked
+   */
+  function stake(
+    uint256[] memory tokenIds
+  ) public nonReentrant whenNotPaused updateReward(msg.sender) {
```

Comment:
You defined this functions (stake, unstake, etc) in interface, so this function can be override function

Reply:
Actually, solidity v0.8.0, we don't need to add `override` keyword, but better to add for code readability